lightbeam/lightbeam.py

```diff
@@ -160,12 +160,33 @@ def get_data_files_for_endpoint(self, endpoint):
         return file_list
 
     # Prunes the list of endpoints down to those for which .jsonl files exist in the config.data_dir
-    def get_endpoints_with_data(self, endpoints):
+    def get_endpoints_with_data(self):
```

Comment:
You're right, they're similar. I built out this `get_endpoints_with_data()` function because it's much more performant to loop over the (presumably relatively small number of) files in `data_dir` and check the name of each, than to loop over hundreds of Resource/Descriptor endpoints and check for existence of several `DATA_FILE_EXTENSIONS` for each.

Probably a better solution would be to have a function that constructs (using this performant method) a `endpoints_with_data` object like
```yaml
schools:
- [data_dir]/schools.jsonl
students:
- [data_dir]/students.jsonl
studentSchoolAssociations:
- [data_dir]/studentSchoolAssociations/1.jsonl
- [data_dir]/studentSchoolAssociations/2.jsonl
```
and then
* `get_endpoints_with_data()` would be `endpoints_with_data.keys()`
* `get_data_files_for_endpoint(endpoint)` would be `endpoints_with_data[endpoint]`

I think this deserves more thought and feedback though - would you be ok with leaving it for a future refactor, @johncmerfeld ?

Reply:
Definitely happy to leave it!